lib/rules/this-in-template.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @fileoverview enforce usage of `this` in template.
+ * @author Armano
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'enforce usage of `this` in template.',
+      category: 'Best Practices',
+      recommended: false
+    },
+    fixable: null,
+    schema: [
+      {
+        enum: ['always', 'never']
+      }
+    ]
+  },
+
+  /**
+   * Creates AST event handlers for this-in-template.
+   *
+   * @param {RuleContext} context - The rule context.
+   * @returns {Object} AST event handlers.
+   */
+  create (context) {
+    const options = context.options[0] !== 'always' ? 'never' : 'always'
+    let scope = {
+      parent: null,
+      nodes: []
+    }
+
+    function validateNever () {
+      return {
+        'VExpressionContainer ThisExpression' (node) {
+          if (options === 'never') {
```

Comment:
I think that it should check whether variables which have the same name as the property name.

```html
<template>
    <div v-for="x of xs">
        {{this.x}} <!-- we cannot remove this `this.`. -->
    </div>
</template>
```

Also, I think that it should check whether the `this` is `MemberExpression#object`.

```html
<template>
    <child :parent="this" /> <!-- we cannot remove this `this`. -->
</template>
```